src/ng2-facebook-sdk.ts

```diff
@@ -86,13 +101,17 @@ export class FacebookService {
     login(options?: FacebookLoginOptions): Promise<FacebookLoginResponse> {
         return new Promise<FacebookLoginResponse>(
             (resolve, reject) => {
-                FB.login((response: FacebookLoginResponse) => {
-                    if(response.authResponse) {
-                        resolve(response);
-                    }else{
-                        reject();
-                    }
-                }, options);
+                try {
+                    FB.login((response: FacebookLoginResponse) => {
+                        if(response.authResponse) {
+                            resolve(response);
+                        }else{
+                            reject();
+                        }
+                    }, options);
+                } catch (err) {
+                    console.error('FB is not defined');
```

Comment:
Why did you assume here that the only error that could happen is `FB is not defined`?